File: backend/src/main/java/com/codestates/question/entity/Question.java
package com.codestates.question.entity;

import com.codestates.answer.entity.Answer;
import com.codestates.user.entity.User;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

//TODO : Entity 작성

@NoArgsConstructor
@Getter
@Setter
@Entity
public class Question {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long questionId;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String body;

    @Column(nullable = false)
    private boolean isAnswered;

    @Column(nullable = false)
    private boolean isAccepted;
    @Column
    private int viewCount;

    @Column
    private int answerCount;

    @Column(nullable = false)
    private LocalDateTime creationDate = LocalDateTime.now();;

    @Column(nullable = false)
    private LocalDateTime lastEditDate = LocalDateTime.now();;

    @OneToMany(mappedBy = "question")
    private List<QuestionTag> tags = new ArrayList<>();

    @ManyToOne
    @JoinColumn(name = "USER_ID")
    private User user;

    @OneToMany(mappedBy = "question")
    private List<Answer> answers;



}

File: frontend/util/api/question.ts
export type Question = {
  user: {
    user_id: number;
    display_name: string;
  };
  question: {
    question_id: number;
    title: string;
    body: string;
    is_answered: boolean;
    is_accepted: boolean;
    view_count: number;
    answer_count: number;
    creation_date: string;
    last_edit_date: string;
    vote: {
      is_vote: number;
      down_vote_count: number;
      up_vote_count: number;
      score: number;
    };
    tags: Array<{ tag_id: number; name: string }>;
  };
  answers: Array<{
    user: {
      user_id: number;
      display_name: string;
    };
    answer_id: number;
    body: string;
    is_accepted: boolean;
    creation_date: string;
    last_edit_date: string;
    is_vote: number;
    score: number;
  }>;
};
